zebra-network/src/peer_set/initialize.rs

```diff
@@ -211,7 +225,27 @@ where
     S: Service<(TcpStream, SocketAddr), Response = peer::Client, Error = BoxError> + Clone,
     S::Future: Send + 'static,
 {
-    let listener = TcpListener::bind(addr).await?;
+    println!("attempting to open listener");
+    let listener_fut = TcpListener::bind(addr);
+    let listener_fut = tokio::time::timeout(Duration::from_secs(3), listener_fut);
+    let listener_result = listener_fut.await;
+
+    let panic_now = |e: &dyn std::fmt::Debug| {
+        println!("Panicking now");
+        panic!(
+            "Opening Zcash network protocol listener {:?} failed: {:?}. \
+             Hint: Check if another zebrad or zcashd process is running. \
+             Try changing the network listen_addr in the Zebra config.",
+            addr, e,
+        );
+    };
+
+    let listener = match listener_result {
+        Ok(Ok(l)) => l,
+        Ok(Err(e)) => panic_now(&e),
+        Err(e) => panic_now(&e),
```

Comment:
Should we provide a different message for the timeout Elapsed error, or do you think it's enough to add it to the panic message?

Reply:
The error debug just says "Elapsed()", but saying "timeout" would be a lot more helpful.